dpe/src/bitmap.rs

```diff
@@ -0,0 +1,78 @@
+/*++
+Licensed under the Apache-2.0 license.
+
+Abstract:
+    Helper struct for controlling and viewing bitmaps.
+--*/
+
+#[repr(C)]
+#[derive(Debug, PartialEq, Eq, Default)]
+pub struct Bitmap(u32);
+
+impl Bitmap {
+    /// Empty flags.
+    pub const fn const_default() -> Bitmap {
+        Bitmap(0)
+    }
+
+    /// Bitmap with only one flag set.
+    pub const fn flag(flag: usize) -> Bitmap {
+        assert!(flag < u32::BITS as usize);
+        Bitmap(1 << flag)
+    }
+
+    /// Get the value of the given flag.
+    pub const fn get(&self, flag: usize) -> bool {
+        assert!(flag < u32::BITS as usize);
```

Comment:
I know this is expected to never happen, but I think we want to avoid panic wherever possible. I wonder if it's possible to make the compiler enforce the values passed in here. Otherwise I think we should return a Result.

Reply:
We could do some enum magic. Let me play around with it and see what I can come up with